deepspeed/profiling/flops_profiler/profiler.py

```diff
@@ -227,16 +234,47 @@ def get_total_duration(self, as_string=False):
         return duration_to_string(total_duration) if as_string else total_duration
 
     def get_total_params(self, as_string=False):
-        """Returns the total parameters of the model.
+        """Returns the total number of parameters stored per rank.
 
         Args:
             as_string (bool, optional): whether to output the parameters as string. Defaults to False.
 
         Returns:
-            The number of parameters in the model.
+            The total number of parameters stored per rank.
+        """
+        total_params = self.get_total_expert_params() + self.get_total_nonexpert_params()
+        return params_to_string(total_params) if as_string else total_params
+
+    def get_total_nonexpert_params(self, as_string=False):
+        """Returns the total non-expert parameters of the model per rank.
+
+        Args:
+            as_string (bool, optional): whether to output the parameters as string. Defaults to False.
+
+        Returns:
+            The number of parameters in the model stored per rank.
         """
         return params_to_string(self.model.__params__) if as_string else self.model.__params__
 
+
+    def get_total_expert_params(self, as_string=False):
+        """Returns the total number of parameters associated with expert layers per rank.
+
+        Args:
+            as_string (bool, optional): whether to output the parameters as string. Defaults to False.
+
+        Returns:
+            The total number of parameters associated with expert layers per rank.
+        """
+        return params_to_string(self.model.__expert_params__) if as_string else self.model.__expert_params__
+
+
+    def is_expert_tensor_parallelism_enabled(self):
+        for _, module in self.model.named_modules():
+            if isinstance(module, MoE) and hasattr(module, 'enable_expert_tensor_parallelism'):
```

Comment:
nit: `return getattr(module, "enable_expert_tensor_parallelism", False)`

Reply:
That wouldn't be the same. It wouldn't go through each module till it found one with the attribute